tests/unittests/Foundation/NSSetTests.mm

```diff
@@ -60,4 +60,24 @@
     }];
 
     ASSERT_OBJCEQ(expectedEvensLessThanFive, actual);
+}
+
+TEST(NSSet, FilterUsingPredicateString) {
+    NSSet* americanMakes = [NSSet setWithObjects:@"Chrysler", @"Ford", @"General Motors", nil];
+    NSPredicate* predicate = [NSPredicate predicateWithFormat:@"SELF BEGINSWITH[c] %@", @"G"];
+
+    NSSet* filteredSet = [americanMakes filteredSetUsingPredicate:predicate];
+
+    ASSERT_NE(nil, filteredSet);
+    ASSERT_EQ(1, [filteredSet count]);
```

Comment:
but it would be a nice to have. adding.

---
In reply to: [74289845](https://github.com/Microsoft/WinObjC/pull/857#discussion_r74289845) [](ancestors = 74289845,74267425)